### ISSUE TITLE
Error: Key [key] has already been defined, when scaling docker swarm service

### ISSUE THREAD

**Issue by bitsofinfo** [containous/traefik#1853]
Traefik 1.3.3

**1. Create docker network**

`docker network create -d overlay --attachable my-net1`

**2. Start traefik**

```
docker service create \
--name traefik \
--constraint=node.role==manager \
--mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock \
--network my-net1 \
--publish 25080:80 \
--publish 25888:8080 \
--publish 25443:443 \
traefik:1.3.3  \
--debug=true \
--logLevel=DEBUG \
--docker \
--docker.swarmmode \
--docker.domain=traefik \
--docker.watch \
--web \
--web.statistics 
```

**3. Start an nginx service**

```
docker service create \
  --name nginx1 \
  --publish :80 \
  --label traefik.protocol=http \
  --label traefik.port=80 \
  --label traefik.test1.frontend.rule='Host:mytest.mydomain1.com' \
  --label traefik.test1.port=80 \
  --label traefik.docker.network=my-net1 \
  --network my-net1 \
  nginx
```

All is good 

```
2017-07-11T14:40:09.116607641Z time="2017-07-11T14:40:09Z" level=debug msg="Configuration received from provider docker: {"backends":{"backend-nginx1-test1":{"servers":{"service":{"url":"http://10.0.9.5:80","weight":0}},"loadBalancer":{"method":"wrr"}}},"frontends":{"frontend-nginx1-test1":{"entryPoints":["http"],"backend":"backend-nginx1-test1","routes":{"service-test1":{"rule":"Host:mytest.mydomain1.com"}},"passHostHeader":true,"priority":0,"basicAuth":[]}}}"
2017-07-11T14:40:09.116639103Z time="2017-07-11T14:40:09Z" level=debug msg="Last docker config received more than 2s, OK"
2017-07-11T14:40:09.116650090Z time="2017-07-11T14:40:09Z" level=debug msg="Creating frontend frontend-nginx1-test1"
2017-07-11T14:40:09.116660578Z time="2017-07-11T14:40:09Z" level=debug msg="Wiring frontend frontend-nginx1-test1 to entryPoint http"
2017-07-11T14:40:09.117118637Z time="2017-07-11T14:40:09Z" level=debug msg="Creating route service-test1 Host:mytest.mydomain1.com"
2017-07-11T14:40:09.117346467Z time="2017-07-11T14:40:09Z" level=debug msg="Creating backend backend-nginx1-test1"
2017-07-11T14:40:09.117490597Z time="2017-07-11T14:40:09Z" level=debug msg="Creating load-balancer wrr"
2017-07-11T14:40:09.117540438Z time="2017-07-11T14:40:09Z" level=debug msg="Creating server service at http://10.0.9.5:80 with weight 0"
2017-07-11T14:40:09.117562112Z time="2017-07-11T14:40:09Z" level=info msg="Server configuration reloaded on :80"
```

If you go to the traefik dashboard at http://localhost:25888 all is good

**4. Scale it**

`docker service scale nginx1=2`

**5. At this point the traefik logs will show the following error:**

```
2017-07-11T14:40:54.112338205Z time="2017-07-11T14:40:54Z" level=debug msg="Filtering container without port and no traefik.port label traefik.1"
2017-07-11T14:40:54.114680336Z time="2017-07-11T14:40:54Z" level=error msg="Near line 23 (last key parsed 'backends.backend-nginx1-test1.servers'): Key 'backends.backend-nginx1-test1.servers.service' has already been defined."
```

Now if you go to the traefik dashboard at http://localhost:25888 there is no change, there should be 2 backends but only one

Related: #1840, #1138

**Comment by ewah** [comment]
+1

**Comment by peloncano** [comment]
+1

**Comment by ldez** [comment]
Remember the gentle way to participate:
- add a "reaction" on the first message of the issue.
- add more useful and detailed information on the subject.
- resolve the issue by making a PR.

**Comment by bitsofinfo** [comment]
Any news on this?

**Comment by traefiker** [comment]
Closed by #2314.
